jax/experimental/jax2tf/jax_export.py

```diff
@@ -215,9 +169,147 @@ def serialize_native(fun_jax: Callable,
       mlir_module=mlir_module,
       mlir_module_serialized=mlir_module_serialized,
       module_kept_var_idx=module_kept_var_idx,
-      xla_call_module_version=xla_call_module_version,
-      dim_args_spec=dim_args_spec
-  )
+      xla_call_module_version=xla_call_module_version)
+
+
+def compute_dim_vars(module: mlir.ir.Module,
+                     args_avals: Sequence[core.ShapedArray]) -> mlir.ir.Module:
+  """Wraps the lowered module with a new "main" that computes the dim vars.
+
+  JAX lowering in presence of shape polymorphism produces a `module` that
+  takes one or more dimension arguments, of type i32 or i64, followed
```

Comment:
"that takes one or more dimension arguments, specified using 0-dimensional tensors of type i32 or i64"

https://github.com/GleasonK/stablehlo/blob/main/docs/spec.md#get_dimension_size